code/modules/xenoarchaeology/traits/activators.dm

```diff
@@ -0,0 +1,664 @@
+		if(!maneater)
+			M.visible_message("<span class='warning'>[AM] bites [M], it didn't quite like the taste!</span>", "<span class='warning'>[AM] bites you!\n[AM] doesn't like that taste!</span>")
+			return FALSE
+		else
+			M.visible_message("<span class='warning'>[AM] bites [M], it loves the taste!</span>", "<span class='warning'>[AM] bites you!\n[AM] loves that taste!</span>")
+			return ..()
+	return FALSE
+
+/datum/xenoartifact_trait/activator/sturdy/hungry/get_dictionary_hint()
+	return list(XENOA_TRAIT_HINT_TWIN, XENOA_TRAIT_HINT_TWIN_VARIANT("only eat food items"))
+
+/datum/xenoartifact_trait/activator/sturdy/hungry/proc/handle_timer()
+	if(bite_timer)
+		deltimer(bite_timer)
+	bite_timer = null
+
+//maneater variant
+/datum/xenoartifact_trait/activator/sturdy/hungry/maneater
+	material_desc = null
+	label_name = "Hungry Δ"
+	label_desc = "Hungry Δ: The artifact seems to be made of a semi-living, hungry, material. This material seems to be triggered by feeding interactions."
+	flags = XENOA_PLASMA_TRAIT | XENOA_URANIUM_TRAIT | XENOA_BANANIUM_TRAIT | XENOA_PEARL_TRAIT
+	maneater = TRUE
+	conductivity = 8
+
+/datum/xenoartifact_trait/activator/sturdy/hungry/maneater/get_dictionary_hint()
+	return list(XENOA_TRAIT_HINT_TWIN, XENOA_TRAIT_HINT_TWIN_VARIANT("eat food items, and mobs"))
+
+/*
+	Edible
+	This trait activates the artifact when it is eaten
+*/
+/datum/xenoartifact_trait/activator/edible
+	material_desc = "edible"
+	label_name = "Edible"
+	label_desc = "Edible: The artifact seems to be made of an edible material. This material seems to be triggered by being consumed."
+	flags = XENOA_BLUESPACE_TRAIT | XENOA_URANIUM_TRAIT | XENOA_BANANIUM_TRAIT | XENOA_PEARL_TRAIT
+	blacklist_traits = list(/datum/xenoartifact_trait/minor/dense)
+	weight = 16
+	incompatabilities = TRAIT_INCOMPATIBLE_MOB | TRAIT_INCOMPATIBLE_STRUCTURE
+	///What reagents does this artifact provide when eaten?
+	var/food_reagents = list(/datum/reagent/consumable/nutriment = INFINITY)
+	///How long does it take us to bite thise?
+	var/bite_time = 4 SECONDS
+	///How many reagents do we get per bite, maximum
+	var/max_bite_reagents = 2
+
+/datum/xenoartifact_trait/activator/edible/New(atom/_parent)
+	. = ..()
+	if(!parent?.parent)
+		return
+	parent.parent.AddComponent(/datum/component/edible,\
+		initial_reagents = food_reagents,\
+		foodtypes = RAW | MEAT | GORE,\
+		volume = INFINITY,\
+		pre_eat = CALLBACK(src, PROC_REF(pre_eat)),\
+		after_eat = CALLBACK(src, PROC_REF(after_eat)),\
+		eat_time = bite_time,\
+		bite_consumption = (max_bite_reagents * (parent.trait_strength/100)))
+	RegisterSignal(parent?.parent, COMSIG_PARENT_ATTACKBY, TYPE_PROC_REF(/datum/xenoartifact_trait/activator, translation_type_b))
+
+/datum/xenoartifact_trait/activator/edible/Destroy(force, ...)
+	. = ..()
+	if(!parent?.parent)
+		return
+	var/datum/component/edible/E = parent.parent.GetComponent(/datum/component/edible)
+	E.RemoveComponent()
```

Comment:
Components are removed on destroy automatically

Reply:
This case covers when the trait is removed / destroyed but the object isn't, which can happen